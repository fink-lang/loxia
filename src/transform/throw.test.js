import {fink2js} from '../testing';


describe('throw', ()=> {
  it('transforms throw', ()=> {
    expect(
      fink2js(`
        foo = bar || throw err('spam')
      `)
    ).toMatchSnapshot();
  });
});
