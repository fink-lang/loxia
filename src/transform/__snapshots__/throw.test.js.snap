// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throw transforms throw 1`] = `
"bar => {
  const ˆvalue_1 = true;

  if (ˆvalue_1 === has_err(bar)) {
    throw err(\`spam\`);
  }

  {
    return bar;
  }
};

Object.assign(module.exports, {});"
`;
